### DISCUSSION TITLE
Getting bond orders when no hydrogens are present.

### DISCUSSION THREAD

**Discussion by jlkjr** [rdkit/rdkit#8095] [General]
If I have a structure file that has neither bond orders nor hydrogens, is there a collection of rdkit operations (C++ API calls) that can both get bond orders (say from 3D coordinate info but without hydrogens, not templates) and add hydrogens?
I've tried calling assignChiralTypesFrom3D, assign StereochemistryFrom3D, followed by addHs, and this doesn't work. I believe determineBondOrders requires hydrogens to be included. In this case I don't have templates for the ligands, so that method can't be used.  Note, the structure doesn't come from the PDB, but from a protein folding program (Boltz-1 in this case). 
Thanks for your help.

**Comment by greglandrum** [comment]
Unfortunately you either need Hs to determine bond orders or bond orders to determine Hs. Any algorithm that tries to figure out both is going to be tricky and very error prone.
